File: server/internal/domain/news/contract.go
package news

import (
	"time"

	"github.com/google/uuid"
)

type ApprovedSource struct {
	ID            int    `json:"id"`
	SourceKey     string `json:"source_key"`
	DisplayName   string `json:"display_name"`
	Kind          string `json:"kind"`
	AllowedOrigin string `json:"allowed_origin"`
	URLTemplate   string `json:"-"`
	IsActive      bool   `json:"is_active"`
}

type Item struct {
	ID           uuid.UUID `json:"id"`
	SourceID     int       `json:"source_id"`
	URL          string    `json:"url"`
	URLHash      string    `json:"url_hash"`
	Title        string    `json:"title"`
	Content      *string   `json:"content,omitempty"`
	PublishedAt  time.Time `json:"published_at"`
	RelatedCoins []string  `json:"related_coins"`
}

File: app/domain/news/contract.py
"""Pure news collection value objects.

Only server-owned :class:`ApprovedSource` values may reach an outbound adapter.
There is deliberately no public-request URL field in the collection command.
"""

from __future__ import annotations

from dataclasses import dataclass, field
from datetime import datetime
from uuid import UUID


@dataclass(frozen=True)
class ApprovedSource:
    id: UUID
    source_key: str
    display_name: str
    kind: str
    allowed_origin: str
    url_template: str
    is_active: bool = True


@dataclass(frozen=True)
class CollectedItem:
    source_id: UUID
    canonical_url: str
    url_hash: str
    title: str
    content: str
    content_hash: str
    published_at: datetime
    related_coins: tuple[str, ...] = field(default_factory=tuple)
    extraction_version: str = "rss-v1"
    tagging_version: str = "aliases-v1"
